File: src/main/java/com/tngtech/jgiven/scenario/state/ReferenceFactory.java
package com.tngtech.jgiven.scenario.state;

import com.intellij.psi.*;
import com.intellij.psi.impl.light.LightMemberReference;

public class ReferenceFactory {
    private final PsiManager manager;

    public ReferenceFactory(PsiManager manager) {
        this.manager = manager;
    }

    PsiReference referenceFor(PsiField field) {
        return new LightMemberReference(manager, field, PsiSubstitutor.EMPTY) {
            @Override
            public PsiElement getElement() {
                return field;
            }
        };
    }

}

File: src/main/java/com/tngtech/jgiven/scenario/state/ReferenceFactory.kt
package com.tngtech.jgiven.scenario.state

import com.intellij.psi.*
import com.intellij.psi.impl.light.LightMemberReference

class ReferenceFactory(private val manager: PsiManager) {

    fun referenceFor(field: PsiField): PsiReference =
            object : LightMemberReference(manager, field, PsiSubstitutor.EMPTY) {
                override fun getElement(): PsiElement = field
            }
}
